### PR TITLE
Toolbar return variables

### LINKED ISSUE #469: Add meaningful return values for toolbar manipulation methods
**Please note we are preparing to participate in Google Code-in, and have reserved this issue for GCI participants - but we'd love to have your help with another one! Please check out https://code.publiclab.org to see more.**

#  Problem

We have toolbar manipulation methods `#addTool` `#removeTool` and `#replaceTool` for `L.DistortableImage.Edit`:

https://github.com/publiclab/Leaflet.DistortableImage/blob/0a162b57aac8be4644a878901567f306c3c006ee/src/edit/DistortableImage.Edit.js#L211-L238

Currently, the return results for them could be more purposeful.

# Solution

I think a good route to go would be to return false on failure like they already do, and the toolbar object itself on success to allow for chaining and more concise code. Open to other suggestions for a return value if you feel they would be better.

To complete:

- [ ] `addTool` needs a meaningful return value on success (undefined currently)
- [ ] `removeTool` needs a meaningful return value on success & failure (undefined currently)
- [ ] `replaceTool` needs a meaningful return value on success (undefined currently)
- [ ] document these return values in the README (just follow the current documentation pattern - see other tool related methods documented for reference)

This task is marked as `medium` because of the wide range of skills it touches on (code, doc, refactoring, etc.).

## Step by Step
- [ ] Claim this issue with a comment here, below, and ask any clarifying questions you need
- [ ] Fork the repository and set it up locally following the main repo README instructions [https://github.com/publiclab/Leaflet.DistortableImage](https://github.com/publiclab/Leaflet.DistortableImage)
 - [ ] Create a new feature branch with a unique name descriptive to the issue
- [ ] Try to fix the issue following the steps above, but even before you're done, you can:
  commit your changes to your branch and start a pull request (see [contributing to Public Lab software](https://publiclab.org/wiki/contributing-to-public-lab-software)) but mark it as "in progress" if you have questions or if you haven't finished
- [ ] Reference this issue in your pull request body
- [ ] Once you submit your pull request, if there's an additional checklist provided for getting it merged, get those boxes checked off. Either way, mention me @sashadev-sky for a review.

##  💬   Get help
If you need any help - here are some options: 

- Comment below
- Join our gitter chat at https://gitter.im/publiclab/publiclab

### PR DESCRIPTION
**Pull request by tsparksh**
Fixes #469 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `grunt`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updates
* [x] @mention the original creator of the issue in a comment below for help or for a review

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by tsparksh** [comment]
Hey @sashadev-sky, Can you review this?

**Comment by sashadev-sky** [comment]
@tsparksh on it!

**Comment by sashadev-sky** [comment]
also you will need to run `$ grunt` to make sure your dist file (`dist/leafletdistortableimage.js`) is updated with your changes and commit it to your PR. I typically run this in the background on another terminal tab while making changes.

if `$ grunt` doesn't update the file you can force it to with `$ grunt concat:dist`

File: src/edit/DistortableImage.Edit.js [hunk 1]
```diff
@@ -175,9 +175,8 @@ L.DistortableImage.Edit = L.Handler.extend({
 
   replaceTool: function(old, next) {
     if (next.baseClass !== 'leaflet-toolbar-icon' || this.hasTool(next)) {
-      return;
+      return false;
```
Comment (sashadev-sky) [inline src/edit/DistortableImage.Edit.js]: For consistency, this method should only return this too

File: README.md [hunk 2]
```diff
@@ -644,17 +644,17 @@ Same as `L.DistortableImage.Edit` but for the collection (`L.DistortableCollecti
 
 <details><summary><code><b>removeTool(<i>action</i> &#60;EditAction>)</b></code></summary>
```
Comment (sashadev-sky) [inline README.md]:
```suggestion
<details><summary><code><b>removeTool(<i>action</i> &#60;EditAction>)</b>: this</code></summary>
```

File: README.md [hunk 3]
```diff
@@ -644,17 +644,17 @@ Same as `L.DistortableImage.Edit` but for the collection (`L.DistortableCollecti
...
   <ul>
-    <li>Removes the passed tool from the control toolbar in runtime.</li>
+    <li>Removes the passed tool from the control toolbar in runtime. Returns the parent object.</li>
```
Comment (sashadev-sky) [inline README.md]:
```suggestion
    <li>Removes the passed tool from the control toolbar in runtime if the tool is present.</li>
```

File: README.md [hunk 4]
```diff
@@ -644,17 +644,17 @@ Same as `L.DistortableImage.Edit` but for the collection (`L.DistortableCollecti
...
     <li>ex: <code>imgGroup.removeTool(Deletes)</code></li>
   </ul>
 </details>
 
 <details><summary><code><b>addTool(<i>action</i> &#60;EditAction>)</b></code></summary>
```
Comment (sashadev-sky) [inline README.md]:
```suggestion
<details><summary><code><b>addTool(<i>action</i> &#60;EditAction>)</b>: this</code></summary>
```

Comment (sashadev-sky) [inline README.md]:
```suggestion
<details><summary><code><b>addTool(<i>action</i> &#60;EditAction>)</b>: this</code></summary>
```

File: README.md [hunk 5]
```diff
@@ -644,17 +644,17 @@ Same as `L.DistortableImage.Edit` but for the collection (`L.DistortableCollecti
...
-<ul><li>Adds the passed tool to the end of the control toolbar in runtime. Returns false if the tool is not available or is already present.</li></ul>
+<ul><li>Adds the passed tool to the end of the control toolbar in runtime. Returns the parent object.</li></ul>
```
Comment (sashadev-sky) [inline README.md]:
```suggestion
<ul><li>Adds the passed tool to the end of the control toolbar in runtime.</li></ul>
```

**Pull request merged**